CIME/XML/generic_xml.py

```diff
@@ -126,8 +126,10 @@ def read(self, infile, schema=None):
             logger.debug("read: {}".format(infile))
             with open(infile, "r", encoding="utf-8") as fd:
                 self.read_fd(fd)
-
-            if schema is not None and self.get_version() > 1.0:
+            version = str(self.get_version())
+            if type(schema) is dict:
+                self.validate_xml_file(infile, schema[version])
+            elif schema is not None and self.get_version() > 1.0:
```

Comment:
As we discussed over lunch, I am happy with this approach. But can you please add some documentation in the docstring for this function describing the possible types for schema (i.e., that it can now be a string(?) or dict, and if the latter, what the keys are)?